scene/main/viewport.cpp

```diff
@@ -1822,7 +1822,7 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 					if (w->is_embedded()) {
 						embedder = w->_get_embedder();
 
-						Transform2D ai = (get_final_transform().affine_inverse() * _get_input_pre_xform()).affine_inverse();
+						Transform2D ai = get_final_transform();
 
 						viewport_pos = ai.xform(mpos) + w->get_position(); // To parent coords.
```

Comment:
```suggestion
						viewport_pos = get_final_transform().xform(mpos) + w->get_position(); // To parent coords.
```

Reply:
I have included this simplification. Had to rebase the PR.